pkg/driver/driver_gcp.go

```diff
@@ -97,12 +97,9 @@ func (d *GCPDriver) Create() (string, string, error) {
 	}
 	instance.Disks = disks
 
-	var metadataItems = []*compute.MetadataItems{
-		{
-			Key:   "user-data",
-			Value: &d.UserData,
-		},
-	}
+	var metadataItems = []*compute.MetadataItems{}
+	metadataItems = append(metadataItems, d.getUserData())
```

Comment:
Would a more appropriate approach be adding another field for startup-script in the `GCPDriver` struct instead of overloading `UserData`? Could both `user-data` and `startup-script` be used when creating a GCP instance?

Reply:
It is possible to use both but at the moment there is no use case to using them both. Currently they are using the same data. This is the data from the OperationSystemConfig which is translated to #cloud-config or script. There are neither flow or controller which can deal with "startup-script". So we can consider "startup-script" as an alias for "user-data" until there is mechanism to use both.

Reply:
IMO this is specific to the cloud-provider/OS, and better be handled at drivers.